public/src/components/ButtonGroup.jsx

```diff
@@ -1,23 +1,54 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 
 const StyledButtonGroup = styled.div`
+  
+ ${({ horizontal, theme }) => (horizontal ? css`
+    button + button {
+      border-radius: 0;
+    }
+    label  + label {
+      border-radius: 0;
+    }
+    button:first-child {
+      border-top-right-radius: 0;
+      border-bottom-right-radius: 0;
+    }
+    label:first-child {
+      border-top-right-radius: 0;
+      border-bottom-right-radius: 0;
+    }
+    & > button:last-of-type {
+      border-top-right-radius: ${theme.borderRadius};
+      border-bottom-right-radius: ${theme.borderRadius};
+    }
+    & > label:last-of-type {
+      border-top-right-radius: ${theme.borderRadius};
+      border-bottom-right-radius: ${theme.borderRadius};
+    }
+ ` : css`
+  flex-direction: column;
   label + label {
     margin-top: .5em;
   }
   label + button {
     margin-top: .5em;
   }
+ `)}
+  display: flex;
   width: 100%;
 `;
 
-export const ButtonGroup = ({ children }) => (
-  <StyledButtonGroup>
-    {children}
-  </StyledButtonGroup>
-);
+export const ButtonGroup = ({ children, horizontal }) => (
```

Comment:
Nit:
```
export const ButtonGroup = (props) => <StyledButtonGroup {...props} />
```

there's no need to be explicit, and now you have to maintain the list of props (aka if you want to add something, you have to add it here & in the consumer)

Reply:
That makes Eslint shit a brick but I'll just disable the rule.